service/storage_proxy.cc

```diff
@@ -2740,7 +2748,24 @@ struct storage_proxy::remote& storage_proxy::remote() {
 }
 
 const struct storage_proxy::remote& storage_proxy::remote() const {
-    return *_remote;
+    if (_remote) {
+        return *_remote;
+    }
+
+    // This error should not appear because the user should not be able to send queries
+    // before `remote` is initialized, and user queries should be drained before `remote`
+    // is destroyed; Scylla code should take care not to perform cluster queries outside
+    // the lifetime of `remote` (it can still perform queries to local tables during
+    // the entire lifetime of `storage_proxy`, which is larger than `remote`).
+    //
+    // If there's a bug though, fail the query instead of crashing Scylla.
+    //
+    // In the future we may want to introduce a 'recovery mode' in which Scylla starts
+    // without contacting the cluster and allows the user to perform local queries (say,
+    // to system tables), then this code path would be expected to happen if the user
+    // tries a remote query in this recovery mode.
+    slogger.error("attempted to access `remote` when it's uninitialized");
+    throw std::runtime_error{"Remote queries not available"};
```

Comment:
`on_internal_error()`, it will take care of the log + throw and adds a bonus backtrace too.